src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -140,15 +165,20 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_copy_ep_put_short,
                  uint64_t remote_addr, uct_rkey_t rkey)
 {
     uct_cuda_copy_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_cuda_copy_iface_t);
-    uct_cuda_copy_stream_t idx   = UCT_CUDA_COPY_STREAM_H2D;
     ucs_status_t status;
+    ucs_memory_info_t src_info;
+    ucs_memory_info_t dst_info;
+    cudaStream_t *stream;
+
+    uct_cuda_copy_memory_detect(iface, buffer, (size_t)length, &src_info);
+    uct_cuda_copy_memory_detect(iface, (void *)remote_addr, (size_t)length,
```

Comment:
doesn't it add overhead to small messages (same for get_short)?

Reply:
Will change this.

Reply:
Will change this.